Core/Core/Planner/CalendarEvent/View/CustomFrequencyComponents/OccurrencesCountInputDialog.swift

```diff
@@ -0,0 +1,80 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2024-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import SwiftUI
+
+private struct OccurrencesCountInputAlertViewModifier: ViewModifier {
+
+    @Binding var isPresented: Bool
+    @StateObject var inputModel: OccurrencesCountInputModel
+
+    init(isPresented: Binding<Bool>, count: Binding<Int>) {
+        self._isPresented = isPresented
+        self._inputModel = StateObject(
+            wrappedValue: OccurrencesCountInputModel(submitted: count)
+        )
+    }
+
+    func body(content: Content) -> some View {
+        content
+            .alert(String(localized: "Number of Occurrences", bundle: .core),
+                   isPresented: $isPresented) {
+
+                TextField("Occurrences",
+                          value: $inputModel.value,
+                          formatter: formatter)
+                .keyboardType(.asciiCapableNumberPad)
+
+                Button(role: .cancel, action: {
+                    isPresented = false
+                }, label: {
+                    Text("Cancel", bundle: .core)
+                })
+
+                Button(action: {
+                    inputModel.submit()
+                    isPresented = false
+                }, label: {
+                    Text("Done", bundle: .core)
+                })
+                .disabled(inputModel.isValid == false)
```

Comment:
Disable behavior for Alert's Actions seems to work only for iOS 17+.
And for it work properly, we need to make sure it is not disabled when onAppear. 
I made few changes on it to get around this limitation. I had to revert back to use `text` binding of `TextField`.
Which is not a thing I was hoping for 😄

Reply:
Thank you! It seems to work fine on iOS17!
On iOS16 if I input something invalid and press Done (as you mentioned, disable is not working pre 17) it will fallback to some value (the first digit of the previous value). My only idea here for iOS16, if it's a number: clamp it. If it's not, fallback to 0. (so the user can't tap Done with some unintended value)